Confus/Confus/Game.cpp

```diff
@@ -180,6 +184,49 @@ namespace Confus
             ClientTeamScore::setTeamScore(ETeamIdentifier::TeamBlue, blueScore);
             std::cout << "Score updated\tRed score: " << redScore << "\t Blue score: " << blueScore << std::endl;
         });
+
+        m_Connection->addFunctionToMap(static_cast<unsigned char>(Networking::EPacketType::EndOfGame), [this](RakNet::Packet* a_Packet) 
+        {
+            ETeamIdentifier a_TeamIdentifier;
+            RakNet::BitStream inputStream(a_Packet->data, a_Packet->length, false);
+            inputStream.IgnoreBytes(sizeof(RakNet::MessageID));
+            inputStream.Read(a_TeamIdentifier);
+
+            auto text = m_Device->getGUIEnvironment()->addStaticText(L"Resetting Game", { 10, 240, 10 + 150, 240 + 80 });
+
+            // Values to calculate time with
+            irr::u32 currentTicks = m_Device->getTimer()->getRealTime();
+            irr::u32 previousTicks = currentTicks;
+            double time = 0;
+            double breakTime = 60;
+            while(time < breakTime && !m_EventManager.IsKeyDown(irr::KEY_SPACE))
+            {
+                // Show text on GUI
+                if(a_TeamIdentifier == ETeamIdentifier::TeamBlue) 
+                {
+                    text->setText((L"Blue team won! \nPress space to restart now! \nRestarting in: " + std::to_wstring(breakTime - time)).c_str());
+                }
+                else if(a_TeamIdentifier == ETeamIdentifier::TeamRed) 
+                {
+                    text->setText((L"Red team won! \nPress space to restart now! \nRestarting in: " + std::to_wstring(breakTime - time)).c_str());
+                }
+
+                // Call render to update GUI text
+                m_Device->run();
+                m_Connection->processPackets();
+                m_Device->getVideoDriver()->beginScene(true, true, irr::video::SColor(255, 100, 100, 100));
+                m_Device->getGUIEnvironment()->drawAll();
+                m_Device->getVideoDriver()->endScene();
+
+                // Calculate time to see if we've been here longer than breaktime
+                previousTicks = currentTicks;
+                currentTicks = m_Device->getTimer()->getRealTime();
+                time += ((currentTicks - previousTicks) / 1000.0);
+                std::this_thread::sleep_for(std::chrono::milliseconds((1 / 60) * 1000));
```

Comment:
Are you sure it's that great of a deal to block the entire game from execution until a timer has finished?